semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -2262,9 +2266,23 @@ and m_for_header a b =
         ~less_is_ok:false a2 b2
   | G.ForClassic _, _
   | G.ForEach _, _
+  | G.MultiForEach _, _
   | G.ForIn _, _ ->
       fail ()
 
+and m_for_each (a1, at, a2) (b1, bt, b2) =
+  m_pattern a1 b1 >>= fun () ->
+  m_tok at bt >>= fun () -> m_expr a2 b2
+
+and m_multi_for_each a b =
+  match (a, b) with
+  | G.FE a1, B.FE b1 -> m_for_each a1 b1
+  | G.FECond (a1, at, a2), B.FECond (b1, bt, b2) ->
+      m_for_each a1 b1 >>= fun () ->
+      m_tok at bt >>= fun () -> m_expr a2 b2
+  | G.FEllipsis _, _ -> return ()
```

Comment:
I don't think this can can happen given the definition of m_list_with_dots, but I guess it does not hurt.